### DISCUSSION TITLE
Help compiling some package as static

### DISCUSSION THREAD

**Discussion by lars18th** [termux/termux-packages#13542] [Q&A]
Hi,

I'm using the docker container to generate packages:
https://github.com/termux/termux-packages/wiki/Building-packages

So, I don't have any problem when executing `./build-package.sh -a arm <some_package>`. And the packages are well generated.
However, I need your help to compile **STATIC** a package that uses CMake.

For example, I see that the package `proot-static` doesn't have any problem because it uses gnu makefiles:
https://github.com/ZhymabekRoman/proot-static

But I need to compile a package (`putty-tools`, a subpackage of `putty`), and I'm failing compiling it static.
I've tested this patch:

[putty-static.patch]
```
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,9 @@
 cmake_minimum_required(VERSION 3.7)
 project(putty LANGUAGES C)
-
+set_property(GLOBAL PROPERTY LINK_SEARCH_START_STATIC ON)
+set_property(GLOBAL PROPERTY LINK_SEARCH_END_STATIC ON)
+set(BUILD_SHARED_LIBRARIES OFF)
+  
 include(cmake/setup.cmake)
 
 # Scan the docs directory first, so that when we start calling
```

But it doesn't generate any static binary.
So please, can you share with me some information about how to modify a Termux package to compile static when it's based on CMake.
Thank you.

**Comment by Grimler91** [comment]
putty-tools programs link against libandroid-glob.so (and no other shared libraries as far as I can see), so update the build recipe to use libandroid-glob.a instead:

```diff
diff --git a/x11-packages/putty/build.sh b/x11-packages/putty/build.sh
index 58d65a74e20b..3ddd19de2377 100644
--- a/x11-packages/putty/build.sh
+++ b/x11-packages/putty/build.sh
@@ -6,7 +6,8 @@ TERMUX_PKG_VERSION=0.78
 TERMUX_PKG_SRCURL=https://the.earth.li/~sgtatham/putty/${TERMUX_PKG_VERSION}/putty-${TERMUX_PKG_VERSION}.tar.gz
 TERMUX_PKG_SHA256=274e01bcac6bd155dfd647b2f18f791b4b17ff313753aa919fcae2e32d34614f
 TERMUX_PKG_DEPENDS="gdk-pixbuf, glib, gtk3, libcairo, libx11, pango"
+TERMUX_PKG_BUILD_DEPENDS="libandroid-glob-static"
 
 termux_step_pre_configure() {
-       LDFLAGS+=" -landroid-glob"
+       LDFLAGS+=" $TERMUX_PREFIX/lib/libandroid-glob.a"
 }
```

For main putty package it is not as easy, we do not even have static libraries of all dependencies.

**Reply by lars18th** [reply]
Hi @Grimler91 ,

After compiling the package with this change in the build compiling...
```
$ ldd psusan
        libm.so => /system/lib/libm.so
        libc.so => /system/lib/libc.so
        ld-android.so => /system/lib/ld-android.so
        libdl.so => /system/lib/libdl.so
```

Therefore the result is not static. Any idea?
Thank you.

**Comment by lars18th** [comment]
Hi @Grimler91 ,

I'm compiling now with this:
```
TERMUX_PKG_BUILD_DEPENDS="libandroid-glob-static"

termux_step_pre_configure() {
	LDFLAGS+=" $TERMUX_PREFIX/lib/libandroid-glob.a"
	LDFLAGS+=" /home/builder/lib/android-ndk-r25b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/lib/x86_64-linux-android/libm.a"
	LDFLAGS+=" /home/builder/lib/android-ndk-r25b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/lib/x86_64-linux-android/libc.a"
	LDFLAGS+=" /home/builder/lib/android-ndk-r25b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/lib/x86_64-linux-android/libdl.a"
}
```

And this generates a bigger binary. However:
```
$ patchelf --print-needed psusan
libm.so
```

So, still is linked with the `libm.so` shared library. And this library is using `libc.so`, `libdl.so` and `ld-android.so`. Therefore, any idea to remove this last dependency?

Thank you for your help.
